components/tree/private/item.jsx

```diff
@@ -36,13 +37,13 @@ const handleSelect = ({ event, props, fromFocus }) => {
 
 const findNextNode = (flattenedNodes, node) => {
 	const nodes = flattenedNodes.map((flattenedNode) => flattenedNode.node);
-	const index = nodes.indexOf(node);
+	const index = findIndex(nodes, (item) => item.id === node.id);
```

Comment:
Consider using lodash's iteratee shorthand here too:
```suggestion
	const index = findIndex(nodes, {'id': node.id});
```